--- a/Messages/Makefile.in
+++ b/Messages/Makefile.in
@@ -37,8 +37,8 @@ $(LANGUAGES_FILE): FORCE
 	done
 
 template $(TEMPLATE_EXTENSION): $(TEMPLATE_FILE)
-$(TEMPLATE_FILE): $(BLD_TOP)$(PGM_DIR)/cmds.auto.h FORCE
-	$(SRC_TOP)mkpot -p "$(PACKAGE_TARNAME)" -n "$(PACKAGE_VERSION)" -b "$(PACKAGE_BUGREPORT)" -c "$(COPYRIGHT_HOLDER)" -o $@ -- $(SRC_TOP) $(BLD_TOP)
+#$(TEMPLATE_FILE): $(BLD_TOP)$(PGM_DIR)/cmds.auto.h FORCE
+#	$(SRC_TOP)mkpot -p "$(PACKAGE_TARNAME)" -n "$(PACKAGE_VERSION)" -b "$(PACKAGE_BUGREPORT)" -c "$(COPYRIGHT_HOLDER)" -o $@ -- $(SRC_TOP) $(BLD_TOP)
 
 sources: $(SOURCE_FILES)
 %.$(SOURCE_EXTENSION): FORCE
