diff --git a/Programs/brlapi_constants.awk b/Programs/brlapi_constants.awk
index 4104996..c73816a 100644
--- a/Programs/brlapi_constants.awk
+++ b/Programs/brlapi_constants.awk
@@ -91,5 +91,5 @@ function writeBrlapiDots() {
   print "#define BRLAPI_DOTS(" arguments ") (\\\n" expression " \\\n)"
 
   print ""
-  writeMacroDefinition("BRLAPI_DOT_CHORD", 0X100, "space key")
+  writeMacroDefinition("BRLAPI_DOT_CHORD", 256, "space key")
 }
